monai/data/grid_dataset.py

```diff
@@ -187,21 +187,25 @@ def __init__(
     ) -> None:
         super().__init__(data=data, transform=None)
         self.patch_iter = patch_iter
-        self.transform = transform
+        self.patch_transform = transform
         self.with_coordinates = with_coordinates
 
     def __iter__(self):
```

Comment:
I'd suggest reducing the whole method a bit more:

```python
def __iter__(self):
    for image in super().__iter__():
        for patch, *others in self.patch_iter(image):  # patch_iter to yield at least 1 item: patch
            out_patch = patch
            if self.patch_transform is not None:
                out_patch = apply_transform(self.patch_transform, patch, map_items=False)

            if self.with_coordinates and len(others) > 0:
                yield out_patch, others[0]
            else:
                yield out_patch
```

Reply:
makes sense, thanks!